data_rentgen/server/utils/lineage_response.py

```diff
@@ -137,3 +151,77 @@ def _get_output_relations(outputs: dict[Any, Output]) -> list[LineageOutputRelat
         relations.append(relation)
 
     return sorted(relations, key=lambda x: (str(x.from_.id), str(x.to.id), x.type))
+
+
+def _get_direct_column_lineage_relations(
+    dataset_relations_by_target_source_ids: dict[tuple, list[ColumnLineage]],
+    column_relations_by_fingerpint: dict[UUID, list[DatasetColumnRelation]],
+) -> list[DirectLineageColumnRelationV1]:
+    relations = []
+    for (target_dataset_id, source_dataset_id), dataset_relations in dataset_relations_by_target_source_ids.items():
+        relation = DirectLineageColumnRelationV1(
+            from_=LineageEntityV1(kind=LineageEntityKindV1.DATASET, id=str(source_dataset_id)),
+            to=LineageEntityV1(kind=LineageEntityKindV1.DATASET, id=str(target_dataset_id)),
+        )
+        fingerprints = {
+            dataset_relation.fingerprint: dataset_relation.created_at for dataset_relation in dataset_relations
+        }
+        for fingerprint, last_used_at in fingerprints.items():
+            fields = {}  # type: ignore[var-annotated]
+            for column_relation in column_relations_by_fingerpint[fingerprint]:
+                if column_relation.type <= ColumnLineageInteractionTypeV1.AGGREGATION_MASKING.value:  # type: ignore[operator]
```

Comment:
column_relation.type is db.models.ColumnLineageInteractionType, and should not be compared with HTTP response schema enum.